101-DDoS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json

```diff
@@ -0,0 +1,111 @@
+            ],
+            "defaultValue": "Free",
+            "metadata": {
+              "description": "Service Tier: Free, Standalone, or PerNode"
+            }
+        },
+        "dataRetention": {
+            "type": "int",
+            "defaultValue": 90,
+            "minValue": 7,
+            "maxValue": 730,
+            "metadata": {
+                "description": "Number of days of retention. Free plans can only have 7 days, Standalone and OMS plans include 30 days for free"
+            }
+        },
+        "location": {
+            "type": "string",
+            "allowedValues": [
+                "East US",
+                "West Europe",
+                "Southeast Asia",
+                "Australia Southeast"
+            ],
+            "defaultValue": "East US"
+        }
+    },
+    "resources": [
+        {
+            "name": "[parameters('omsWorkspaceName')]",
+            "type": "Microsoft.OperationalInsights/workspaces",
+            "apiVersion": "2017-04-26-preview",
+            "location": "[parameters('location')]",
+            "tags": "[parameters('tags')]",
+            "properties": {
+              "sku": {
+                "name": "[parameters('sku')]"
+              },
+              "retention": "[parameters('dataRetention')]"
+            }
+        },
+        {
+            "type": "Microsoft.OperationsManagement/solutions",
+            "name": "[concat(parameters('omsSolutionsName')[copyIndex()], '(', parameters('omsWorkspaceName'), ')')]",
+            "apiVersion": "2015-11-01-preview",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[parameters('omsWorkspaceName')]"
+            ],
+            "copy": {
+                "name": "copy-oms-solutions",
+                "count": "[length(parameters('omsSolutionsName'))]"
+            },
+            "plan": {
+                "name": "[concat(parameters('omsSolutionsName')[copyIndex()], '(', parameters('omsWorkspaceName'), ')')]",
+                "product": "[concat('OMSGallery/', parameters('omsSolutionsName')[copyIndex()])]",
+                "promotionCode": "",
+                "publisher": "Microsoft"
+            },
+            "properties": {
+                "workspaceResourceId":  "[resourceId('Microsoft.OperationalInsights/workspaces', parameters('omsWorkspaceName'))]"
+            }
+        },
+        {
+            "type": "Microsoft.OperationalInsights/workspaces/datasources",
+            "kind": "AzureActivityLog",
+            "name": "[concat(parameters('omsWorkspaceName'), '/', subscription().subscriptionId)]",
+            "apiVersion": "2015-11-01-preview",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[parameters('omsWorkspaceName')]"
+            ],
+            "properties": {
+                "linkedResourceId": "[concat(subscription().Id, '/providers/microsoft.insights/eventTypes/management')]"
```

Comment:
use resourceID

Reply:
hi @bmoore-msft , we could not figure out how can we use resourceID function to get this resource id. I looked after some references in the quickstart itself to find out that the same method was used to get it.
https://github.com/Azure/azure-quickstart-templates/blob/master/azmgmt-demo/nestedtemplates/omsWorkspace.json
https://github.com/Azure/azure-quickstart-templates/blob/master/oms-all-deploy/nestedtemplates/omsWorkspace.json
please let us know if you have any suggestion on how to use it.